cpp/src/arrow/csv/writer.cc

```diff
@@ -406,7 +405,7 @@ class CSVWriterImpl : public ipc::RecordBatchWriter {
     std::vector<std::unique_ptr<ColumnPopulator>> populators(schema->num_fields());
     for (int col = 0; col < schema->num_fields(); col++) {
       const std::string& end_chars =
-          col < schema->num_fields() - 1 ? kStrComma : options.eol;
+          col < schema->num_fields() - 1 ? std::string(1,options.delimiter) : options.eol;
```

Comment:
While the compiler will likely do this for us, we should probably hoist the string construction outside of the loop.

Reply:
Yes, right, that's better.